### PR TITLE
Add GetDisabledTaskExecutors Api + scaler handling on disabled TEs

### PR DESCRIPTION
**Pull request by Andyz26**
### Context
Adding api to query disabled TEs.
[update]
Updating the logic for:
* Add condition for disabled TEs metrics to only include active (registered) TEs.
* Include disabled TEs in auto scaler usage so disabled TEs won't pollute the scale-up calculation.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

**Review by fdc-ntflx — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 1]
```diff
@@ -216,6 +216,12 @@ protected Route constructRoutes() {
                     (clusterName) -> pathEndOrSingleSlash(() -> concat(
                         get(() -> mkTaskExecutorsRoute(getClusterID(clusterName), (rc, req) -> rc.getBusyTaskExecutors(req.getAttributes())))))
                 ),
+                // /{}/getBusyTaskExecutors
```
Comment (hmitnflx) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:219]:
```suggestion
                // /{}/getDisabledTaskExecutors
```

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java
Comment (hmitnflx) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:1]: Can you add the new endpoint as a comment at the top of the file as well?
* /api/v1/resourceClusters/{}/getDisabledTaskExecutors (GET)

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterAkkaImpl.java [hunk 2]
```diff
@@ -118,6 +119,16 @@ public CompletableFuture<List<TaskExecutorID>> getBusyTaskExecutors(Map<String,
             .thenApply(l -> l.getTaskExecutors());
     }
 
+    @Override
+    public CompletableFuture<List<TaskExecutorID>> getDisabledTaskExecutors(Map<String, String> attributes) {
+        return Patterns.ask(
+                resourceClusterManagerActor,
```
Comment (hmitnflx) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterAkkaImpl.java:125]: Can you add an akka handler for this endpoint in `ResourceClustersManagerActor` as well?

`.match(GetDisabledTaskExecutorsRequest.class, req -> getRCActor(req.getClusterID()).forward(req, context()))`

**Review by hmitnflx — APPROVED** [review]

**Review by sundargates — APPROVED** [review]
LGTM!

**Pull request merged**